### PR TITLE
remove native object access and add explicit ipython_repr_mimebundle API

### LINKED ISSUE #15: task: clarify naming of `repr_mimedisplay` -> ipython display

### PR DESCRIPTION
**Pull request by alisterburt**
finally got around to finishing this! no more native, v satisfying :) should make the multiple backend adaptors work easier 

will close #15

### CODE REVIEW

File: src/microvis/backend/vispy/_canvas.py [hunk 1]
```diff
@@ -69,3 +69,8 @@ def _vis_render(
             region=region, size=size, bgcolor=bgcolor, crop=crop, alpha=alpha
         )
         return cast("np.ndarray", data)
+
+    def _vis_get_ipython_mimebundle(
+        self, *args: Any, **kwargs: Any
+    ) -> dict | tuple[dict, dict]:
+        return self._vis_get_native()._repr_mimebundle_(*args, **kwargs)  # type: ignore
```
Comment (alisterburt) [inline src/microvis/backend/vispy/_canvas.py:76]: type ignore here, was complaining about returning Any in a typed context - wasn't sure if it was solvable so left

File: src/microvis/core/view.py [hunk 2]
```diff
@@ -138,10 +138,10 @@ def show(self) -> Canvas:
...
         """Add any node to the scene."""
         self.scene.add(node)
-        if self.camera.has_adaptor:
-            # FIXME!: put this vispy specific API elsewhere
-            # i guess we need a reset_range type API
-            self.camera.native.set_range(margin=0)
+        # if self.camera.has_adaptor:
+        #     # FIXME!: put this vispy specific API elsewhere
+        #     # i guess we need a reset_range type API
+        #     self.camera.native.set_range(margin=0)
```
Comment (alisterburt) [inline src/microvis/core/view.py:144]: this used the native API directly - have commented out for now and tests are still passing but as you say, we need a fix. What does the reset_range API you envision mean? I'm never exactly sure what 'range' refers to, is it some notion of the 'viewing area'?

File: src/microvis/core/canvas.py [hunk 3]
```diff
@@ -137,9 +141,12 @@ def _repr_mimebundle_(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
...
         This defer to the native object's _repr_mimebundle_ method if it exists.
         Allowing different backends to support Jupyter or other rich display.
+
+        https://ipython.readthedocs.io/en/stable/config/integrating.html#more-powerful-methods
         """
-        if hasattr(self.native, "_repr_mimebundle_"):
-            return cast(dict, self.native._repr_mimebundle_(*args, **kwargs))
+        adaptor = self.backend_adaptor()
+        if hasattr(adaptor, "_viz_get_ipython_mimebundle"):
+            return adaptor._viz_get_ipython_mimebundle(*args, **kwargs)  # type: ignore
```
Comment (alisterburt) [inline src/microvis/core/canvas.py:149]: Another type error I wasn't sure how to deal with

Comment (alisterburt) [reply src/microvis/core/canvas.py:149]: Same thing, returning Any when the method has a return type

File: src/microvis/core/view.py (hunk 2 above)
Comment (tlambert03) [reply src/microvis/core/view.py:144]: Yeah, the api needs some consideration.  The effect of this call to set_range(margin=0) here is essentially to set the zoom (and position?) such that the image exactly fills the viewport.  Not sure we need a backend call for this at all actually, it seems it should be possible with zoom and center position alone right?  (This was just a short term patch to get me something to look at in the canvas)

This also reminds me, our zoom and position docstring/definitions should be explicit about the units/spaces in which they are defined.  (Ie, what is a zoom of 1)

**Comment by tlambert03** [comment]
General comment here, I do think we should keep some aspect of retrieving native objects, but once this is done, we could make sure we don't actually _use_ it internally, and we could a name (like react) that indicates lack of support, something like "dangerously_get_backend_object"

**Comment by alisterburt** [comment]
yeah sure that's totally reasonable, internal use is the thing which potentially complicates life - I can add it back into this PR (will come with a backend arg)

**Comment by tlambert03** [comment]
ok, i think this PR is good to go, but by commenting out that vispy-specific set_range bit, it does make things relatively unusable for now.  That's fine with me (it was always a temporary hack)... the next thing would need to be nailing down our internal camera model.  Could start by looking at both the pygfx and vispy cameras (and others if we can find them), with the hopes of finding unifying features.

**Comment by alisterburt** [comment]
Thanks for finishing this off! Merging

**Comment by alisterburt** [comment]
Agree on the camera model - I'll take a look and open an issue to discuss

**Comment by alisterburt** [comment]
merge conflicts fixed, see #37 for camera API thoughts

**Pull request merged**